dcos/mesos.py

```diff
@@ -273,22 +382,61 @@ class Framework(object):
 
     :param framework: framework properties
     :type framework: dict
+    :param master: framework's master
+    :type master: Master
     """
 
-    def __init__(self, framework):
+    def __init__(self, framework, master):
         self._framework = framework
+        self._master = master
+        self._tasks = {}  # id->Task map
+
+    def task(self, task_id):
+        """Returns a task by id
+
+        :param task_id: the task's id
+        :type task_id: str
+        :returns: the task
+        :rtype: Task
+        """
+
+        for task in _merge(self._framework, ['tasks', 'completed_tasks']):
+            if task['id'] == task_id:
+                return self._task_obj(task)
+        return None
+
+    def _task_obj(self, task):
+        """Returns the Task object corresponding to the provided `task`
+        dict.  Creates it if it doesn't exist already.
+
+        :param task: task
+        :type task: dict
+        :returns: Task
+        :rtype: Task
+        """
+
+        if task['id'] not in self._tasks:
+            self._tasks[task['id']] = Task(task, self._master)
+        return self._tasks[task['id']]
 
     def dict(self):
         return self._framework
 
     def __getitem__(self, name):
+        """Support the framework[attr] syntax
+
+        :param name: attribute to get
+        :type name: str
+        :returns: the value for this attribute in the underlying
+                  framework dictionary
+        :rtype: object
+        """
+
         return self._framework[name]
 
 
 class Task(object):
-    """Mesos Task Model.  Created from the Task objects sent in master's
-    state.json, which is in turn created from mesos' Task protobuf
-    object.
+    """Mesos Task Model.
```

Comment:
"Mesos Task Model"?  This is the class that models a mesos task.  It's backed by the task entry in state.json